### ISSUE TITLE
[bug] stepper icons - accessibility - icon hidden text read as a separated tab in screen-reader

### ISSUE THREAD

**Issue by jumbolu** [angular/components#29864]
Hello, 

I am questioned by the accessibility testers while using Stepper.  

1. The icons in stepper-header are considered as decorative icons (aria-hidden="true") except "edit" and "done". I am suggested the "number" icons are meaningful, and should be read by the screen-reader as "Step 1", "Step 2", etc. 

2. The indicator icons (edit and done) are read as a separated tab in the screen-reader:
 - "tab Editable" then followed by "tab matStepLabel" 
 - "tab Completed" then followed by "tab matStepLabel"  
![Image](https://github.com/user-attachments/assets/65166385-c509-4333-9521-fc9386728573)

Any comments or suggestions to address these accessibility concerns? Thank you!

**Comment by crisbeto** [comment]
From what I can tell, these don't show up in the accessibility tree anymore.
